### ISSUE TITLE
insta review fails with a hidden snapshot directory (starting with '.')

### ISSUE THREAD

**Issue by ngryman** [mitsuhiko/insta#326]
### What happened?

Hi,

I'm using a custom directory for my snapshot located at the root of my project at `$CARGO_MANIFEST_DIR/.snapshots`. When running my tests, snapshots are generated correctly. However, `cargo insta review` fails to locate them.

Changing the directory to a non-hidden one; removing the `.` prefix; works fine.

## Theory

After quickly scanning the source I think it may come from these lines: https://github.com/mitsuhiko/insta/blob/ab93e8da849d9bc577007357beb8fbcf694b4c21/cargo-insta/src/cargo.rs#L257-L258

From the [WalkDir documentation](https://docs.rs/ignore/latest/ignore/struct.WalkBuilder.html#method.standard_filters), it seems that the `standard_filters` re-enables the `hidden` option. So it might be the case that it's overriding the previous `hidden(false)` option in the context of reviewing snapshots.

As a side note, my `.snapshots` directory is not git ignored so my issue must come from the `hidden` option, and not the `git_ignore` one enabled by `standard_filters`.

### Reproduction steps

1. Set a hidden directory as the snapshot directory with `with_settings!`.
2. Run your test.
3. Check that snapshots are generated properly.
4. Run `cargo insta review`.
5. The following message appears: `done: no snapshots to review`.

### What did you expect?

I expect `cargo insta review` to properly locate the snapshot directory.

**Comment by mitsuhiko** [comment]
This is intentional behavior. You need to pass `--no-ignore` to `cargo insta review`.

**Comment by mitsuhiko** [comment]
With the next version you will be able to pass `--include-hidden` to have insta walk into hidden folders.

**Comment by ngryman** [comment]
@mitsuhiko Thanks for the quick reply & fix!

**Comment by ngryman** [comment]
Hi @mitsuhiko,

I just updated to `1.24.0` and my issue is resolved but not as expected.

**tl;dr `--include-ignored` works but `--include-hidden` has no effect.**

---

I have the same use case as before, which is:

1. I want my snapshot directories to be `.snapshots`.
2. I don't ignore `.snapshots` in any of the files listed in [WalkBuilder's standard filters](https://docs.rs/ignore/latest/ignore/struct.WalkBuilder.html#method.standard_filters).

### Current behavior

:one: When I run `cargo-insta` with `--include-hidden`, I get the following:

![Screen Shot 2023-01-04 at 11 22 29 AM](https://user-images.githubusercontent.com/892048/210536057-aa390f66-6d3b-415a-970d-2acad31b1348.png)

:two: When I run `cargo-insta` with `--include-ignored`, it works.

### Expected behavior

In my use case, I would expect :one: to have the same behavior than :two:.

The error message in :one: should not appear as it advises me to use the flag I did use.
